src/App.tsx

```diff
@@ -1,26 +1,235 @@
+import cn from 'classnames';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  createTodos,
+  deleteTodos,
+  getTodos,
+  todosServiceErrorText,
+  USER_ID,
+} from './api/TodoServise';
+import { Todo } from './types/Todo';
+import { Filter, FilterStatus } from './components/Filter';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoList } from './components/TodoList';
+import { NewTodo } from './components/CreateTodo';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const shouldShowUserWarning = !USER_ID;
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<FilterStatus>(FilterStatus.All);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const focusedInput = useRef<HTMLInputElement>(null);
+
+  const [todoTitle, setTodoTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [creating, setIsCreating] = useState(false);
+  const [processing, setProcessing] = useState<number[]>([]);
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      try {
+        setIsLoading(true);
+        setErrorMessage(null);
+        const loadedTodos = await getTodos();
+
+        setTodos(loadedTodos);
+      } catch {
+        setErrorMessage(todosServiceErrorText.Unable_to_load_todos);
+        setTimeout(() => setErrorMessage(null), 3000);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchTodos();
+  }, []);
+
+  useEffect(() => {
+    if (!isLoading && !creating && focusedInput.current) {
+      focusedInput.current.focus();
+    }
+  }, [isLoading, creating]);
+
+  const getFilteredTodos = (todosToFilter: Todo[], filters: FilterStatus) => {
+    switch (filters) {
+      case FilterStatus.Active:
+        return todosToFilter.filter(todo => !todo.completed);
+      case FilterStatus.Completed:
+        return todosToFilter.filter(todo => todo.completed);
+      case FilterStatus.All:
+      default:
+        return todosToFilter;
+    }
+  };
+
+  const handleAddTodo = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const title = todoTitle.trim();
+
+    if (!title) {
+      setErrorMessage(todosServiceErrorText.title_should_not_be_empty);
+      setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+      focusedInput.current?.focus();
+
+      return;
+    }
+
+    setErrorMessage(null);
+    setIsCreating(true);
+
+    const temp: Todo = {
+      id: 0,
+      userId: USER_ID,
+      title,
+      completed: false,
+    };
+
+    setTempTodo(temp);
+
+    try {
+      const created = await createTodos(title);
+
+      setTodos(prev => [...prev, created]);
+      setTodoTitle('');
+    } catch {
+      setErrorMessage(todosServiceErrorText.Unable_to_add_a_todo);
+      setTimeout(() => {
+        setErrorMessage(null);
+      }, 3000);
```

Comment:
This pattern is repeated many times

How's about making a function for this?

```suggestion
      const showError = errorText => {
         setErrorMessage(todosServiceErrorText.Unable_to_add_a_todo);
         setTimeout(() => {
           setErrorMessage(null);
         }, 3000);
      }


      showError(todosServiceErrorText.Unable_to_add_a_todo);
```